docs/reference/samplers.rst

```diff
@@ -15,6 +15,11 @@ distributions defined by the problem.
     :local:
     :depth: 1
 
+These samplers are non-blocking: the returned :class:`~dimod.SampleSet` is constructed
+from a :class:`~concurrent.futures.Future`-like object that is resolved on the first
+read of any of its properties; for example, by printing the results or querying its
```

Comment:
Actually, `.done()` is non-blocking -- it immediately returns if sampleset has been resolved or now.

OTOH, printing will block, as well as data access methods/properties like `.info`, `.record`, `.vartype`, `.first`, or even `len(sampleset)`.

Reply:
It says the samplers are non-blocking. The confusion comes from the example incorrectly adding "or querying its ...". It should rather say separately that you can query the status with the done method.